api/views/personal_info.py

```diff
@@ -1,9 +1,25 @@
-from rest_framework import viewsets
+from rest_framework import status, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.serializers import BaseSerializer
 
 from api.models import PersonalInfo
-from api.serializers import PersonalInfoSerializer
+from api.serializers import PersonalInfoCheckExistenceSerializer, PersonalInfoSerializer
 
 
 class PersonalInfoViewSet(viewsets.ModelViewSet):
     queryset = PersonalInfo.objects.all()
-    serializer_class = PersonalInfoSerializer
+
+    def get_serializer_class(self) -> type[BaseSerializer]:
+        if self.action == "check_existence":
+            return PersonalInfoCheckExistenceSerializer
+        return PersonalInfoSerializer
+
+    @action(detail=False, methods=["post"])
```

Comment:
So we're saying that the bot is going to try and create a personal info item (method `POST`) and fails if it exists? 
I was thinking that bot would send a `GET` request with name and email and if they exist, it will abort the registration. But if they do not exist, it will collect more data from the user (and only then send the whole personal info to the back end). So the existence check is not meant to be at the time when _all_ personal info is collected. I don't want to waste the (duplicate) user's time :)

Or maybe I'm misunderstanding something.

Reply:
I think you're misunderstanding :)

We have one endpoint for creating PersonalInfo (POST /api/personal_info/) - it receives all the data once, so you need to get all data from user before sending the request.

I've made this endpoint to check unique constraint violation at the beginning of user registration process in bot. So when you have only first name, last name and email, you could make a request to check only this data, not all fields.

I use POST (even if it doesn't create anything) in security purposes. GET params are not encrypted in https, they are stored in logs, etc. etc., so better to send it as request body.

Reply:
OK, that's good to know, thanks.